server/cmwell-ws/app/logic/services/ServiceDefinition.scala

```diff
@@ -0,0 +1,31 @@
+/**
+  * Copyright 2015 Thomson Reuters
+  *
+  * Licensed under the Apache License, Version 2.0 (the “License”); you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *   http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+  * an “AS IS” BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package logic.services
+
+sealed trait ServiceDefinition {
+  val route: String
+}
+
+case class RedirectionService(override val route: String, srcPattern: String, replaceFunc: String => String) extends ServiceDefinition
+
+sealed abstract class LogicService(override val route: String, logicPath: String,
+                                   args: Array[String], returnType: ServiceReturnType) extends ServiceDefinition
+
+case class Source(override val route: String, logicPath: String,
+                  args: Array[String], returnType: ServiceReturnType) extends LogicService(route, logicPath, args, returnType)
+case class Binary(override val route: String, logicPath: String,
+                  args: Array[String], returnType: ServiceReturnType) extends LogicService(route, logicPath, args, returnType)
+
+case class ServiceReturnType(infotonType: String, mimeType: Option[String])
```

Comment:
We talked about commenting it out.
Maybe it's best to add something like "Feature under construction"